platforms/nuttx/src/bootloader/nxp/imxrt_common/main.c

```diff
@@ -407,40 +422,22 @@ flash_func_sector_size(unsigned sector)
  * */
 locate_code(".ramfunc")
 void
-flash_func_erase_sector(unsigned sector)
+flash_func_erase_sector(unsigned sector, bool force)
 {
-
 	if (sector > BOARD_FLASH_SECTORS || (int)sector < BOARD_FIRST_FLASH_SECTOR_TO_ERASE) {
 		return;
 	}
 
-	/* blank-check the sector */
-	const uint32_t bytes_per_sector =  flash_func_sector_size(sector);
-	uint32_t *address = (uint32_t *)(IMXRT_FLEXSPI1_CIPHER_BASE + (sector * bytes_per_sector));
-	const uint32_t uint32_per_sector =  bytes_per_sector / sizeof(*address);
-	bool blank = true;
+	if (force || !flash_func_is_sector_blank(sector)) {
```

Comment:
so this reads `if (force || flash_func_is_sector_blank(sector) != 0)`

Reply:
Ok!